gyft.py

```diff
@@ -126,38 +126,44 @@ def merge_slots(in_dict):
     for time in timetable_dict[day]:
         timetable_dict[day][time].append(" ")
         course_code = timetable_dict[day][time][0]
-        course_dept = course_code[:2]
+        course_dept = course_code[:2] if course_code[:2] != 'EP' else 'RJ' # special case -> EP courses are under RJ department, not EP department
         if course_dept not in courses.keys():
             courses[course_dept] = {}
         if course_code not in courses[course_dept].keys():
             courses[course_dept][course_code] = ''
 
 # scraping course names deptwise
 for dept in courses.keys():
-    DEPT_URL = COURSES_URL.format(dept)
-    r = s.get(DEPT_URL, headers=headers)
-    soup = bs(r.text, 'html.parser')
-    parentTable = soup.find('table', {'id': 'disptab'})
+    try: 
+        DEPT_URL = COURSES_URL.format(dept)
+        r = s.get(DEPT_URL, headers=headers)
+        soup = bs(r.text, 'html.parser')
+        parentTable = soup.find('table', {'id': 'disptab'})
 
-    rows = parentTable.find_all('tr')
+        rows = parentTable.find_all('tr')
 
-    for row in rows[1:]:
-        if 'bgcolor' in row.attrs:
-            continue 
-        cells = row.find_all('td')  
-        course_code = cells[0].text.strip()
-        course_name = cells[1].text.strip()
+        for row in rows[1:]:
+            if 'bgcolor' in row.attrs:
+                continue 
+            cells = row.find_all('td')  
+            course_code = cells[0].text.strip()
+            course_name = cells[1].text.strip()
 
-        if course_code in courses[dept]:
-            courses[dept][course_code] = course_name
-            logging.info(" {} - {}".format(course_code, course_name))
+
+            if course_code in courses[dept]:
+                courses[dept][course_code] = course_name
+                logging.info(" {} - {}".format(course_code, course_name))
+    except Exception as e:
+        logging.error("Error while scraping course names for department {}".format(dept))
+        logging.error(e)
+        logging.error("You can manually add the course names while executing generate_ics.py")
```

Comment:
Slight nitpicking: Please add a space after `ERROR:root:`

Also. Should we have some blank lines before and after the error lines?

![image](https://github.com/metakgp/gyft/assets/75582834/34676160-3c64-464c-bb3a-08d87140a5e0)

Reply:
Yeah, I suppose a blank line before and after would look better, I'll add it